src/settings/pages/PickListValueSettings.js

```diff
@@ -0,0 +1,127 @@
+    }).isRequired,
+    resources: PropTypes.shape({
+      categories: PropTypes.arrayOf({
+        id: PropTypes.string,
+        desc: PropTypes.string,
+        values: PropTypes.arrayOf({
+          id: PropTypes.string,
+          value: PropTypes.string,
+          label: PropTypes.string,
+        }),
+      }),
+    }),
+    mutator: PropTypes.shape({
+      categories: PropTypes.shape({
+        GET: PropTypes.func.isRequired,
+        reset: PropTypes.func.isRequired,
+      }),
+    })
+  };
+
+  constructor(props) {
+    super(props);
+    this.connectedControlledVocab = props.stripes.connect(ControlledVocab);
+
+    this.state = {
+      categoryId: null,
+    };
+  }
+
+  /**
+   * Refresh lookup tables when the component mounts. Fetches in the manifest
+   * will only run once (in the constructor) but because this object may be
+   * unmounted/remounted without being destroyed/recreated, the lookup tables
+   * will be stale if they change between unmounting/remounting.
+   */
+  componentDidMount() {
+    ['categories'].forEach(i => {
+      this.props.mutator[i].reset();
+      this.props.mutator[i].GET();
+    });
+  }
+
+  onChangeCategory = (e) => {
+    this.setState({ categoryId: e.target.value });
+  }
+
+  renderCategories(intl) {
+    const categories = [];
+    categories.push(
+      { value: 'empty', label: intl.formatMessage({ id: 'ui-licenses.pickListSelect' }) }
+    );
+
+    (((this.props.resources.categories || {}).records || []).forEach(i => {
+      categories.push(
+        { value: i.id, label: i.desc }
+      );
+    }));
+
+    if (!categories.length) {
+      return <div />;
+    }
+
+    return categories;
+  }
+
+  renderRowFilter(intl) {
+    return (
+      <Select
+        dataOptions={this.renderCategories(intl)}
+        id="categorySelect"
+        label={intl.formatMessage({ id: 'ui-licenses.pickList' })}
+        name="categorySelect"
+        onChange={this.onChangeCategory}
+      />
+    );
+  }
+
+  render() {
+    return (
+      <IntlConsumer>
+        {intl => (
+          <this.connectedControlledVocab
+            {...this.props}
+            actuatorType="refdata"
+            baseUrl={`licenses/refdata/${this.state.categoryId}`}
+            columnMapping={{
+              label: intl.formatMessage({ id: 'ui-licenses.headings.value' }),
+              actions: intl.formatMessage({ id: 'ui-licenses.actions' }),
+            }}
+        // We have to unset the dataKey to prevent the props.resources in
+        // <ControlledVocab> from being overwritten by the props.resources here.
+            dataKey={undefined}
+            hiddenFields={['lastUpdated', 'numberOfObjects']}
+            id="pick-list-values"
+            label={intl.formatMessage({ id: 'ui-licenses.settings.pickListValues' })}
+            listSuppressor={() => !this.state.categoryId}
+            nameKey="label"
+            preCreateHook={(item) => Object.assign({}, item, { id: this.state.categoryId })}
```

Comment:
In terms of style:
```suggestion
            preCreateHook={(item) => ({ ...item, id: this.state.categoryId })}
```

But really, are you sure about this? You're setting the `id` on a record you're about to create to be the same as the id of the category that's selected? Why? `id` is always supposed to be unique.

Reply:
No, I'm not sure, I was just copying.